paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonNew.kt

```diff
@@ -0,0 +1,381 @@
+            .fillMaxWidth()
+            .height(IntrinsicSize.Max),
+        transitionSpec = {
+            fadeIn(animationSpec = tween(FADE_ANIMATION_DURATION)) togetherWith
+                fadeOut(animationSpec = tween(FADE_ANIMATION_DURATION, FADE_OUT_ANIMATION_DELAY))
+        },
+        label = "ContentAnimation",
+    ) { completed ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 8.dp)
+        ) {
+            if (completed) {
+                Complete(onProcessingCompleted)
+            } else {
+                Incomplete(
+                    text = when (processingState) {
+                        is PrimaryButtonProcessingState.Idle -> label
+                        else -> stringResource(R.string.stripe_paymentsheet_primary_button_processing)
+                    },
+                    enabled = enabled,
+                    processing = processingState !is PrimaryButtonProcessingState.Idle,
+                    locked = locked,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun BoxScope.Incomplete(
+    text: String,
+    processing: Boolean,
+    locked: Boolean,
+    enabled: Boolean,
+) {
+    val context = LocalContext.current
+    val style = StripeTheme.primaryButtonStyle
+
+    val textStyle = TextStyle(
+        fontFamily = style.typography.fontFamily?.let { fontFamilyId ->
+            FontFamily(Font(fontFamilyId))
+        } ?: FontFamily.Default,
+        fontSize = style.typography.fontSize,
+        fontWeight = FontWeight.Medium,
+    )
+
+    val onBackground = Color(style.getOnBackgroundColor(context)).copy(LocalContentAlpha.current)
+
+    Text(
+        text = text,
+        color = onBackground,
+        style = textStyle,
+        modifier = Modifier.align(Alignment.Center),
+    )
+
+    if (processing) {
+        LoadingIndicator(
+            color = onBackground,
+            modifier = Modifier.align(Alignment.CenterEnd),
+        )
+    } else if (locked) {
+        Icon(
+            painter = painterResource(
+                id = R.drawable.stripe_ic_paymentsheet_googlepay_primary_button_lock
+            ),
+            tint = onBackground.run {
+                if (enabled) {
+                    this
+                } else {
+                    copy(TEXT_SUBDUED_VISIBILITY)
+                }
+            },
+            contentDescription = "lock",
+            modifier = Modifier.align(Alignment.CenterEnd),
+        )
+    }
+}
+
+@Composable
+private fun BoxScope.Complete(
+    onProcessingCompleted: () -> Unit
+) {
+    val inInspectionMode = LocalInspectionMode.current
+
+    val tint = if (isSystemInDarkTheme()) {
+        Color.Black
+    } else {
+        Color.White
+    }
+
+    var animationCompleted by remember {
+        mutableStateOf(false)
+    }
+
+    var alignment by remember {
+        mutableFloatStateOf(
+            if (inInspectionMode) {
+                0f
```

Comment:
0 is centered and 1 is right-aligned, correct? Can we create some constants for this to add this context?